rust/src/parser.rs

```diff
@@ -0,0 +1,308 @@
+/* Copyright (C) 2017 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Pierre Chifflier  <chifflier@wzdftpd.net>
+
+//! Parser registration functions and common interface
+
+use core::{DetectEngineState,Flow,AppLayerEventType,AppLayerDecoderEvents,STREAM_TOCLIENT,STREAM_TOSERVER};
+
+use log::*;
+
+use libc;
+use libc::{c_void,c_char,c_int};
+use std::ffi::CString;
+
+
+// Application layer protocol identifiers (app-layer-protos.h)
+
+pub const ALPROTO_UNKNOWN : AppProto = 0;
+pub const ALPROTO_FAILED : AppProto = 20;
```

Comment:
this seems like asking for trouble, as adding new protos in the C side will possibly change it. Can we somehow set it like you did in ALPROTO_NTP?

Reply:
oh, I missed that one.
updated as well